### PR TITLE
flake8: make entire repository pass `spack flake8 -a`

### PR DESCRIPTION
**Pull request by tgamblin**
- [x] Fix trailing whitespace missed by the bug described in #12755.

- [x] Fix other style issues that have crept in over time (this can happen when flake8 adds new checks with new versions)

### CODE REVIEW

**Comment by tgamblin** [comment]
Still resisting the urge to [blacken](https://github.com/psf/black) everything, but if I can ever get it to work automatically on PRs w/o affecting users, Spack will be blackened.

File: var/spack/repos/builtin/packages/athena/package.py [hunk 1]
```diff
@@ -21,17 +21,60 @@ class Athena(AutotoolsPackage):
...
+                'shkset2d',
+                'shkset3d',
+                'shu-osher',
+                'strat',
+                'streaming2d_multi',
+                'streaming2d_single',
+                'streaming3d_multi',
+                'streaming3d_single',
+                'twoibw'
+            ])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/athena/package.py]: Is there any other way we can have this comply with flake8 and not take up 50+ lines?

Comment (tgamblin) [reply var/spack/repos/builtin/packages/athena/package.py]: Yeah but is it so bad?  One thing per line makes it easier to read, and minimizes diffs if you change it.  I *think* this is also what black would do to it.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/athena/package.py]: I guess it's fine... It sounds like black will make the `import_modules` lists pretty long too.

**Pull request merged**